src/libopensc/card-idprime.c

```diff
@@ -196,8 +196,9 @@ static int idprime_process_index(sc_card_t *card, idprime_private_data_t *priv,
 			&& (memcmp(&start[12], "mscp", 5) == 0)) {
 			new_object.fd++;
 			if (card->type == SC_CARD_TYPE_IDPRIME_V2) {
-				/* The key reference starts from 0x11 */
-				new_object.key_reference = 0x10 + new_object.fd;
+				/* The key reference starts from 0x11 and increments by the key id (ASCII) */
+				int key_id = start[8] - '0';
```

Comment:
I suppose an integer underflow isn't critical here, right?

Reply:
the value is passed to the set security environment APDU so it would always be better to be safe. I will add a safeguard.